texar/modules/classifiers/xlnet_classifier.py

```diff
@@ -0,0 +1,298 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+XLNet Classifier.
+"""
+
+from typing import Any, Dict, Iterable, List, Optional, Tuple
+
+import itertools
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from texar.hyperparams import HParams
+from texar.modules.classifiers.classifier_base import ClassifierBase
+from texar.modules.encoders.xlnet_encoder import XLNetEncoder
+from texar.utils.utils import dict_fetch
+
+
+__all__ = [
+    "XLNetClassifier",
+]
+
+
+class XLNetClassifier(ClassifierBase):
+    r"""Classifier based on XLNet modules.
+
+    Arguments are the same as in
+    :class:`~texar.modules.XLNetEncoder`.
+
+    Args:
+        pretrained_model_name (optional): a str with the name
+            of a pre-trained model to load selected in the list of:
+            `xlnet-base-cased`, `xlnet-large-cased`.
+            If `None`, will use the model name in :attr:`hparams`.
+        cache_dir (optional): the path to a folder in which the
+            pre-trained models will be cached. If `None` (default),
+            a default directory will be used.
+        hparams (dict or HParams, optional): Hyperparameters. Missing
+            hyperparameters will be set to default values. See
+            :meth:`default_hparams` for the hyperparameter structure
+            and default values.
+    """
+
+    def __init__(self,
+                 pretrained_model_name: Optional[str] = None,
+                 cache_dir: Optional[str] = None,
+                 hparams=None):
+
+        super().__init__(hparams)
```

Comment:
Let's explicitly write `super().__init__(hparams=hparams)` to ensure super constructor works in case of multi-inheritance.